data_animation.py

```diff
@@ -0,0 +1,207 @@
+        self.axis.set_xlabel("Time (s)")
+        starting_time_actual = STARTING_TIME - config_ani.START_TIME_GRAPHICAL_OFFSET
+        ending_time_actual = ENDING_TIME - config_ani.START_TIME_GRAPHICAL_OFFSET
+        self.axis.set_xlim(starting_time_actual, ending_time_actual)
+        self.axis.set_ylim(min, max)
+
+
+# Each data item that will have a curve
+class PlotItem:
+    def __init__(self, display_name, csv_colum, axis_class, color, filtered=True):
+        self.name = display_name
+        self.csv_colum = csv_colum
+        self.axis = axis_class.axis
+        self.color = color
+
+        self.data = m[csv_colum]
+        if filtered:
+            self.data = savgol_filter(self.data, FILTERING_STRENGTH, 2)
+        self.plotted_data = []  # Starts off empty and grows over time
+
+        # Create the empty line object that will be filled out as the animiation runs
+        self.line = self.axis.plot([], [], label=f"{display_name}", color=color)[0]
+
+    """a tick pushing the data forward one recording frame, and this might happen multiple 
+    times per animation frame"""
+    def tick(self):
+        self.plotted_data.append(self.data[time_cursor])
+        plotted_time_modified = [ts - config_ani.START_TIME_GRAPHICAL_OFFSET for ts in plotted_time]
+        self.line.set_data(plotted_time_modified, self.plotted_data)
+        return self.line
+
+
+# Import graphing settings from config
+unit_axies = {}
+
+for axis_name, axis_config in config_ani.AXIES.items():
+    unit_axies[axis_name] = PlotAxis(
+        axis_config["display_name"],
+        axis_config["unit"],
+        axis_config["min"],
+        axis_config["max"],
+        axis_config["primary"],
+    )
+
+lines = []
+
+for item_name, item_config in config_ani.PLOT_ITEMS.items():
+    lines.append(
+        PlotItem(
+            item_name,
+            item_config["csv_colum"],
+            unit_axies[item_config["axis"]],
+            item_config["color"],
+            item_config["filtered"],
+        )
+    )
+
+figure_title = config_ani.FIGURE_TITLE
+if config_ani.FIGURE_GRID:
+    plt.grid()
+plt.title(config_ani.FIGURE_TITLE, fontdict=config_ani.GLOBAL_FONT)
+
+# Generate one together legend for each of the curves
+handles = []
+labels = []
+for axis in unit_axies.values():
+    h, l = axis.axis.get_legend_handles_labels()
+    l_with_units = [f"{line} ({axis.unit})" for line in l]
+    handles += h
+    labels += l_with_units
+
+plt.legend(handles, labels, loc=config_ani.LEGEND_LOCATION, prop=config_ani.GLOBAL_FONT)
+
+
+# The method for this timing to work is that the animation runs at a certain framerate 
+# (converted to a time interval), and a time length
+# For each frame, it'll step up the running time variable by the time interval
+# It must then add all the data who's recorded timestamp is smaller than or equal to the
+# running time variable
+def animate(frame_number):
+    global time_cursor
+    global time
```

Comment:
Oof using `global` should be avoided. Also to organize this code a bit more, can we refactor and put all the animation-related stuff in a class? That will solve both problems at once

Reply:
Yeah I actually started doing a cleaning pass to address this issue. Stay tuned for a git push to clean this up.

Reply:
Implemented (quite well may I say) by @Artem-Sotnikov in [856c997](https://github.com/waterloo-rocketry/data-video-visualization/pull/1/commits/856c99790e4501d3754d2777ea7af2ee0f3d0687).